pages/settings/activity.vue

```diff
@@ -0,0 +1,155 @@
+  <v-container>
+    <v-row>
+      <v-col cols="12" :offset-md="1" md="5" class="pt-0 pb-4">
+        <v-list two-line class="mb-2">
+          <v-list-item-group>
+            <v-list-item
+              v-for="file in logFiles"
+              :key="file.Name"
+              :href="getLogFileLink(file.Name)"
+              target="_blank"
+            >
+              <v-list-item-avatar>
+                <v-icon>mdi-file</v-icon>
+              </v-list-item-avatar>
+              <v-list-item-content>
+                <v-list-item-title v-text="file.Name" />
+                <v-list-item-subtitle
+                  v-text="getFormattedLogDate(file.DateModified)"
+                />
+              </v-list-item-content>
+              <v-list-item-action>
+                <v-icon>mdi-chevron-right</v-icon>
+              </v-list-item-action>
+            </v-list-item>
+          </v-list-item-group>
+        </v-list>
+      </v-col>
+      <v-col cols="12" md="5" class="pt-0 pb-4">
+        <v-list two-line class="mb-2" disabled>
+          <v-list-item-group>
+            <v-list-item v-for="activity in activityList" :key="activity.Id">
+              <v-list-item-avatar
+                :color="getColorFromSeverity(activity.Severity)"
+              >
+                <v-icon>mdi-shield-account-outline</v-icon>
+              </v-list-item-avatar>
+              <v-list-item-content>
+                <v-list-item-title v-text="activity.Name" />
+                <v-list-item-subtitle v-text="activity.ShortOverview" />
+              </v-list-item-content>
+              <v-list-item-action>
+                <v-list-item-subtitle
+                  class="text-capitalize-first-letter"
+                  v-text="getFormattedActivityDate(activity.Date)"
+                />
+              </v-list-item-action>
+            </v-list-item>
+          </v-list-item-group>
+        </v-list>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapActions } from 'vuex';
+import colors from 'vuetify/lib/util/colors';
+import { ActivityLogEntry, LogFile, LogLevel } from '~/api';
+import htmlHelper from '~/mixins/htmlHelper';
+
+export default Vue.extend({
+  mixins: [htmlHelper],
+  middleware: 'adminMiddleware',
+  data() {
+    return {
+      activityList: [] as ActivityLogEntry[],
+      logFiles: [] as LogFile[]
+    };
+  },
+  head() {
+    return {
+      title: this.$store.state.page.title
+    };
+  },
+  async beforeMount() {
+    this.setPageTitle({ title: this.$t('activityLogs') });
+    const minDate = new Date();
+    minDate.setTime(minDate.getTime() - 7 * 24 * 60 * 60 * 1000);
+    this.activityList =
+      (await this.$api.activityLog.getLogEntries({ minDate })).data.Items || [];
+    this.logFiles = (await this.$api.system.getServerLogs()).data;
+  },
+  methods: {
+    ...mapActions('page', ['setPageTitle']),
+    getColorFromSeverity(severity: LogLevel): string {
```

Comment:
This is a good case for a computed property

Reply:
didnt get you :confused:

Reply:
Ah ignore me, I didn't realize this function was taking parameters.

Compared to methods:

> Instead of a computed property, we can define the same function as a method. For the end result, the two approaches are indeed exactly the same. However, the difference is that computed properties are cached based on their reactive dependencies. A computed property will only re-evaluate when some of its reactive dependencies have changed. This means as long as message has not changed, multiple access to the reversedMessage computed property will immediately return the previously computed result without having to run the function again.



https://vuejs.org/v2/guide/computed.html